GetStream/stream-chat-android
# Fix marking read after sending a message

closes #2725 

### 🎯 Goal

Fix marking read after sending a message

### 🛠 Implementation details

Using `{ it.createdAt ?: it.createdLocallyAt }` for storing date when marking a message as read is problematic.  We don't support marking read in offline and`createdLocallyAt != null` only when the message is sent by the current user.
The problem happens when `createdLocallyAt` differs from `createdAt` (for example - the device's time is wrongly set) because the date will be assigned to `lastMarkReadEvent = lastMessageDate` and `markRead` function won't be executed if `lastMarkReadEvent <= lastMessageDate`.

### 🧪 Testing

Try to:
1. Change the device's time - move it 1h forward
2. Enable airplane mode and send a message
3. Disable airplane mode and wait for synchronization
4. Go to ChannelList
5. Send a message from different device to the same channel
6. Observe that unread count disappears when clicking on this channel (on the first device)

Test the flow on `develop` and this branch - observe that on `develop` 6. fails.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
